com.unity.multiplayer.mlapi/Runtime/Core/BatchingUtil.cs

```diff
@@ -0,0 +1,187 @@
+                default:
+                    break;
+            }
+            return ret;
+        }
+
+        /// <summary>
+        /// QueueItem
+        /// Add a FrameQueueItem to be sent
+        /// </summary>queueItem
+        /// <param name="item">the threshold in bytes</param>
+        public void QueueItem(in FrameQueueItem item)
+        {
+            foreach (ulong clientId in GetTargetList(item))
+            {
+                // todo: actually queue and buffer. For now, the dict contains just one entry !!!
+
+                if (!SendDict.ContainsKey(clientId))
+                {
+                    SendDict[clientId] = new SendStream();
+                    SendDict[clientId].Item = item;
+                    SendDict[clientId].Writer = new PooledBitWriter(SendDict[clientId].Stream);
+
+                    SendDict[clientId].Writer.WriteBit(false); // Encrypted
+                    SendDict[clientId].Writer.WriteBit(false); // Authenticated
+
+                    switch (item.QueueItemType)
+                    {
+                        case RPCQueueManager.QueueItemType.ServerRPC:
+                            SendDict[clientId].Writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
+                            break;
+                        case RPCQueueManager.QueueItemType.ClientRPC:
+                            SendDict[clientId].Writer.WriteBits(MLAPIConstants.MLAPI_STD_CLIENT_RPC, 6); // MessageType
+                            break;
+                    }
+                }
+
+                // write the amounts of bytes that are coming up
+                PushLength(item.MessageData.Count, ref SendDict[clientId].Writer);
+
+                // write the message to send
+                // todo: is there a faster alternative to .ToArray()
+                SendDict[clientId].Writer.WriteBytes(item.MessageData.ToArray(), item.MessageData.Count);
+
+                ProfilerStatManager.bytesSent.Record((int)item.MessageData.Count);
+                ProfilerStatManager.rpcsSent.Record();
+            }
+        }
+
+        public delegate void SendCallbackType(ulong clientId, SendStream messageStream);
+        public delegate void ReceiveCallbackType(BitStream messageStream, MLAPI.RPCQueueManager.QueueItemType messageType, ulong clientId, float time);
+
+        /// <summary>
+        /// SendItems
+        /// Send any batch of RPC that are of length above threshold
+        /// </summary>
+        /// <param name="threshold"> the threshold in bytes</param>
+        /// <param name="sendCallback"> the function to call for sending the batch</param>
+        public void SendItems(int threshold, SendCallbackType sendCallback)
+        {
+            List<ulong> sent = new List<ulong>();
+
+            foreach (KeyValuePair<ulong, SendStream> entry in SendDict)
+            {
+                // read the queued message
+                using PooledBitWriter writer = SendDict[entry.Key].Writer;
```

Comment:
minor note: apparently this `using var XXX` syntax is not supported on all compiler, I got some compile errors on Unity2020.2 so it might worth convering it to `using (var XXX) { ... }` block instead

Reply:
Good to know!  With the writer becoming a member, the whole using statement is removed.